plugins/plugin-testing-java/plugin-testing-testng/che-plugin-testing-testng-server/src/main/java/org/eclipse/che/plugin/testing/testng/server/TestNGRunner.java

```diff
@@ -40,8 +71,114 @@ public TestResult execute(Map<String, String> testParameters) throws Exception {
     }
 
     @Override
+    @Nullable
     public ProcessHandler execute(TestExecutionContext context) {
+        String projectPath = context.getProjectPath();
+        IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject(projectPath);
+        IJavaProject javaProject = JavaModelManager.getJavaModelManager().getJavaModel().getJavaProject(project);
+        if (javaProject.exists()) {
+            return startTestProcess(javaProject, context);
+        }
+
+        return null;
+    }
+
+    private ProcessHandler startTestProcess(IJavaProject javaProject, TestExecutionContext context) {
+        File suiteFile = createSuite(context, javaProject);
+        if (suiteFile == null) {
+            throw new RuntimeException("Can't create TestNG suite xml file.");
+        }
+
+        JavaParameters parameters = new JavaParameters();
+        parameters.setJavaExecutable("java");
+        parameters.setMainClassName("org.testng.CheTestNGLauncher");
+        parameters.setWorkingDirectory("/tmp");
```

Comment:
@davidfestal Each test framework will have own runner where working directory can by set.
In this case working directory is "/tmp" because TestNG framework generates test-output directory which contains some generated files. So, if this dir will be the root directory of the project, we'll see it in the project explorer.